AsyncDisplayKitTests/ASEditableTextNodeTests.m

```diff
@@ -47,27 +47,105 @@ - (void)setUp
+  XCTAssertTrue(editableTextNode.textView.returnKeyType == UIReturnKeyGo,                              @"textView's returnKeyType should be UIReturnKeyGo.");
+  XCTAssertTrue(editableTextNode.textView.enablesReturnKeyAutomatically == YES,                        @"textView's enablesReturnKeyAutomatically should be YES.");
+  
+  ASEditableTextNode *secureEditableTextNode = [[ASEditableTextNode alloc] init];
+  secureEditableTextNode.secureTextEntry = YES;
+  
+  XCTAssertTrue(secureEditableTextNode.textView.secureTextEntry == YES,                                @"textView's isSecureTextEntry should be YES.");
+}
+
+- (void)testUITextInputTraitsChangeTraitAfterViewLoaded
+{
+  // UITextView ignores any values set on the first 3 properties below if secureTextEntry is enabled.
+  // Because of this UIKit behavior, we'll test secure entry seperately
+  ASEditableTextNode *editableTextNode = [[ASEditableTextNode alloc] init];
+
+  editableTextNode.textView.autocapitalizationType = UITextAutocapitalizationTypeWords;
+  editableTextNode.textView.autocorrectionType = UITextAutocorrectionTypeYes;
+  editableTextNode.textView.spellCheckingType = UITextSpellCheckingTypeYes;
+  editableTextNode.textView.keyboardType = UIKeyboardTypeTwitter;
+  editableTextNode.textView.keyboardAppearance = UIKeyboardAppearanceDark;
+  editableTextNode.textView.returnKeyType = UIReturnKeyGo;
+  editableTextNode.textView.enablesReturnKeyAutomatically = YES;
+  
+  XCTAssertTrue(editableTextNode.textView.autocapitalizationType == UITextAutocapitalizationTypeWords, @"textView's autocapitalizationType should be UITextAutocapitalizationTypeAllCharacters.");
+  XCTAssertTrue(editableTextNode.textView.autocorrectionType == UITextAutocorrectionTypeYes,           @"textView's autocorrectionType should be UITextAutocorrectionTypeYes.");
+  XCTAssertTrue(editableTextNode.textView.spellCheckingType == UITextSpellCheckingTypeYes,             @"textView's spellCheckingType should be UITextSpellCheckingTypeYes.");
+  XCTAssertTrue(editableTextNode.textView.keyboardType == UIKeyboardTypeTwitter,                       @"textView's keyboardType should be UIKeyboardTypeTwitter.");
+  XCTAssertTrue(editableTextNode.textView.keyboardAppearance == UIKeyboardAppearanceDark,              @"textView's keyboardAppearance should be UIKeyboardAppearanceDark.");
+  XCTAssertTrue(editableTextNode.textView.returnKeyType == UIReturnKeyGo,                              @"textView's returnKeyType should be UIReturnKeyGo.");
+  XCTAssertTrue(editableTextNode.textView.enablesReturnKeyAutomatically == YES,                        @"textView's enablesReturnKeyAutomatically should be YES.");
+  
+  ASEditableTextNode *secureEditableTextNode = [[ASEditableTextNode alloc] init];
+  secureEditableTextNode.textView.secureTextEntry = YES;
+  
+  XCTAssertTrue(secureEditableTextNode.textView.secureTextEntry == YES,                                @"textView's isSecureTextEntry should be YES.");
+}
```

Comment:
potentially a little overkill...but it did point out the UIKit issue with secureText :)